pf/tfbridge/main.go

```diff
@@ -44,6 +44,16 @@ func Main(ctx context.Context, pkg string, prov ProviderInfo, meta ProviderMetad
 			}
 			return pp.(*provider).marshalProviderInfo(ctx), nil
 		})
+
+	// Apply deferred autonaming, if any.
```

Comment:
Deferred AutoNaming is a bit ugly, but it's possible. It could also help us outside of PF as a general strategy... make sure AutoNaming is applied *at the end*. 

If we do not want that then another good alternative is to first take a change that eliminates PF-custom ProviderInfo and unifies everything so that even in PF case, `prov.P` is populated with a schema-only provider, then SetAutonaming does not need custom conditionals for it.